### PR TITLE
refactor event processing logic

### PR DESCRIPTION
**Pull request by tirumaraiselvan**
No-op code refactor/cleanup

### Description

What component does this PR affect? 

- [x] Server
- [ ] Console
- [ ] CLI
- [ ] Docs
- [ ] Community Content
- [ ] Build System

Requires changes from other components? If yes, please mark the components:

- [ ] Server
- [ ] Console
- [ ] CLI
- [ ] Docs
- [ ] Community Content
- [ ] Build System

### Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update
- [ ] Community content
- [x] No-op
### Checklist:

- [x] I have read the **[contributing guide](https://github.com/hasura/graphql-engine/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
- [ ] This change requires a change in the documentation. 
- [x] I have updated the documentation accordingly.
- [x] I have added required tests.

### CODE REVIEW

File: server/src-lib/Hasura/Events/Lib.hs [hunk 1]
```diff
@@ -199,53 +199,101 @@ processEvent
...
-  (logger:: HLogger) <- asks getter
-  res <- tryWebhook logenv pool e
-  finally <- either errorFn successFn res
-  liftIO $ either (logQErr logger) (void.return) finally
+  cacheRef::CacheRef <- asks getter
+  cache <- liftIO $ readIORef cacheRef
+  let meti = getEventTriggerInfoFromEvent cache e
+  case meti of
+    Nothing -> do
+      logQErr $ err500 Unexpected "table or event-trigger not found in schema cache"
```
Comment (tirumaraiselvan) [inline server/src-lib/Hasura/Events/Lib.hs]: throw http error here

File: server/src-lib/Hasura/Events/Lib.hs [hunk 2]
```diff
@@ -398,13 +358,67 @@ tryWebhook logenv pool e = do
...
+          else modifyTVar' c (+1)
+
+      eitherResp <- runHTTP req  (Just (ExtraContext createdAt eventId))
+
+      -- decrement counter once http is done
+      liftIO $ atomically $ do
+        let EventEngineCtx _ c _ _ = eeCtx
+        modifyTVar' c (\v -> v - 1)
+
+      case eitherResp of
```
Comment (tirumaraiselvan) [inline server/src-lib/Hasura/Events/Lib.hs]: is there a one liner for this?

File: server/src-lib/Hasura/Events/Lib.hs [hunk 3]
```diff
@@ -199,107 +199,200 @@ processEvent
      )
   => LogEnvHeaders -> Q.PGPool -> Event -> m ()
 processEvent logenv pool e = do
-  (logger:: HLogger) <- asks getter
-  res <- tryWebhook logenv pool e
-  finally <- either errorFn successFn res
-  liftIO $ either (logQErr logger) (void.return) finally
+  cacheRef::CacheRef <- asks getter
```
Comment (0x777) [inline server/src-lib/Hasura/Events/Lib.hs]: You don't need these type annotations

File: server/src-lib/Hasura/Events/Lib.hs [hunk 4]
```diff
@@ -199,107 +199,200 @@ processEvent
...
+  cache <- liftIO $ readIORef cacheRef
+  let meti = getEventTriggerInfoFromEvent cache e
+  case meti of
```
Comment (0x777) [inline server/src-lib/Hasura/Events/Lib.hs]: Use `onNothing` from `GraphQL.Utils`.

File: server/src-lib/Hasura/Events/Lib.hs [hunk 5]
```diff
@@ -199,107 +199,200 @@ processEvent
...
+isClientError :: Int -> Bool
+isClientError status = status >= 1000
+
+mkMaybe :: [a] -> Maybe [a]
+mkMaybe [] = Nothing
+mkMaybe x  = Just x
+
+logQErr :: ( MonadReader r m, MonadIO m,  Has HLogger r) => QErr -> m ()
+logQErr err = do
+  (logger:: HLogger) <- asks getter
```
Comment (0x777) [inline server/src-lib/Hasura/Events/Lib.hs]: Same. You don't need these annotations.

File: server/src-lib/Hasura/Events/Lib.hs (hunk 2 above)
Comment (0x777) [reply server/src-lib/Hasura/Events/Lib.hs]: Use `onLeft` from `GraphQL.Utils`. Also please move `onNothing`, `onJust`, `onLeft` from `GraphQL.Utils` to `Hasura.Prelude`

File: server/src-lib/Hasura/Events/Lib.hs (hunk 5 above)
Comment (tirumaraiselvan) [reply server/src-lib/Hasura/Events/Lib.hs]: In some other context, these variables were not resolving to correct type without annotation. But they seem to be fine now.

**Pull request merged**